<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
          "DTD/xhtml1-strict.dtd">
<html>
  <head>
    <title>testtools.testcase.ExpectedException : API documentation</title>
    <meta content="text/html;charset=utf-8" http-equiv="Content-Type" />
    <link href="apidocs.css" type="text/css" rel="stylesheet" />
    
    
  </head>
  <body>
    <h1 class="class">t.t.ExpectedException : class documentation</h1>
    <p>
      <span id="part">Part of <a href="testtools.html">testtools</a>.<a href="testtools.testcase.html">testcase</a></span>
      
      <a href="classIndex.html#testtools.testcase.ExpectedException">View In Hierarchy</a>
    </p>
    <div>
      
    </div>
    <div class="docstring">
      <div><p>A context manager to handle expected exceptions.</p>
<blockquote>
<dl class="rst-docutils">
<dt>def test_foo(self):</dt>
<dd><dl class="rst-first rst-last rst-docutils">
<dt>with ExpectedException(ValueError, 'fo.*'):</dt>
<dd>raise ValueError('foo')</dd>
</dl>
</dd>
</dl>
</blockquote>
<p>will pass.  If the raised exception has a type other than the specified
type, it will be re-raised.  If it has a 'str()' that does not match the
given regular expression, an AssertionError will be raised.  If no
exception is raised, an AssertionError will be raised.</p><table class="fieldTable"></table></div>
    </div>

    
    <div id="splitTables">
      <table class="children sortable" id="id140">
  
  
  <tr class="method">
    
    
    <td>Method</td>
    <td><a href="testtools.testcase.ExpectedException.html#__init__">__init__</a></td>
    <td><span>Construct an <a href="testtools.testcase.ExpectedException.html"><code>ExpectedException</code></a>.</span></td>
  </tr><tr class="method">
    
    
    <td>Method</td>
    <td><a href="testtools.testcase.ExpectedException.html#__enter__">__enter__</a></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="method">
    
    
    <td>Method</td>
    <td><a href="testtools.testcase.ExpectedException.html#__exit__">__exit__</a></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr>
</table>
      
    </div>
    
    

    <div class="function">
  <a name="testtools.testcase.ExpectedException.__init__">
    
  </a>
  <a name="__init__">
    
  </a>
  <div class="functionHeader">
    
    def
    __init__(self, exc_type, value_re=None, msg=None):
    
  </div>
  <div class="docstring functionBody">
    
    <div>Construct an <a href="testtools.testcase.ExpectedException.html"><code>ExpectedException</code></a>.<table class="fieldTable"><tr class="fieldStart"><td class="fieldName">Parameters</td><td class="fieldArg">exc_type</td><td>The type of exception to expect.</td></tr><tr><td></td><td class="fieldArg">value_re</td><td>A regular expression to match against the
'str()' of the raised exception.</td></tr><tr><td></td><td class="fieldArg">msg</td><td>An optional message explaining the failure.</td></tr></table></div>
  </div>
</div><div class="function">
  <a name="testtools.testcase.ExpectedException.__enter__">
    
  </a>
  <a name="__enter__">
    
  </a>
  <div class="functionHeader">
    
    def
    __enter__(self):
    
  </div>
  <div class="docstring functionBody">
    
    <div class="undocumented">Undocumented</div>
  </div>
</div><div class="function">
  <a name="testtools.testcase.ExpectedException.__exit__">
    
  </a>
  <a name="__exit__">
    
  </a>
  <div class="functionHeader">
    
    def
    __exit__(self, exc_type, exc_value, traceback):
    
  </div>
  <div class="docstring functionBody">
    
    <div class="undocumented">Undocumented</div>
  </div>
</div>
    <address>
      <a href="index.html">API Documentation</a> for <a href="https://github.com/testing-cabal/testtools">testtools</a>, generated by <a href="https://launchpad.net/pydoctor/">pydoctor</a> at 2015-05-10 23:16:58.
    </address>
  </body>
</html>